frontend/lib/src/components/elements/PageLink/PageLink.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+import { EmojiIcon } from "@streamlit/lib/src/components/shared/Icon"
+import { Placement } from "@streamlit/lib/src/components/shared/Tooltip"
+import { PageLink as PageLinkProto } from "@streamlit/lib/src/proto"
+import { BaseButtonTooltip } from "@streamlit/lib/src/components/shared/BaseButton"
+
+import { LibContext } from "@streamlit/lib/src/components/core/LibContext"
+import IsSidebarContext from "@streamlit/lib/src/components/core/IsSidebarContext"
+
+import {
+  StyledNavLink,
+  StyledNavLinkText,
+  StyledNavLinkContainer,
+} from "./styled-components"
+
+export interface Props {
+  disabled: boolean
+  element: PageLinkProto
+  width: number
+}
+
+function checkContainerWidthDefault(
+  useContainerWidth: string,
+  isInSidebar: boolean
+): boolean {
+  if (useContainerWidth === "" && isInSidebar) {
+    return true
+  } else if (useContainerWidth === "" && !isInSidebar) {
+    return false
+  }
+  return useContainerWidth === "True" ? true : false
+}
+
+function PageLink(props: Props): ReactElement {
+  const { onPageChange, currentPageScriptHash } = React.useContext(LibContext)
+  const isInSidebar = React.useContext(IsSidebarContext)
+
+  const { disabled, element, width } = props
+  const style = { width }
+
+  const useContainerWidth = checkContainerWidthDefault(
+    element.useContainerWidth,
+    isInSidebar
+  )
+  const isActive = currentPageScriptHash === element.pageScriptHash
+
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+    if (element.external) {
+      // External Page Link
+      if (disabled) {
+        e.preventDefault()
+      }
+    } else {
+      // MPA Page Link
+      e.preventDefault()
+      if (!disabled) {
+        onPageChange(element.pageScriptHash as string)
+      }
+    }
+  }
+
+  return (
+    <div
+      className="row-widget stPageLink"
+      data-testid="stPageLink"
+      style={style}
+    >
+      <BaseButtonTooltip help={element.help} placement={Placement.TOP_RIGHT}>
+        <StyledNavLinkContainer>
+          <StyledNavLink
+            data-testid="stPageNavLink"
```

Comment:
nit: even if it reads strange, should this be `stPageLinkNavLink`? as `stPageLink` is the "parent" and not `stPage`